xds/internal/resolver/xds_resolver_test.go

```diff
@@ -197,20 +225,17 @@ func (s) TestResolverBuilder(t *testing.T) {
 			}
 
 			r, err := builder.Build(test.target, newTestClientConn(), test.buildOpts)
-			if (err != nil) != test.wantErr {
+			if gotErr, wantErr := err != nil, test.wantErr != ""; gotErr != wantErr {
```

Comment:
really really nit: At first this seemed a little confusing to me. But it made sense. 
We are trying to check if gotErr and wantErr are either both Nil or not nil. 

can we rename gotErr and wantErr to something like `gotNilErr` and `wantNilErr` ? 

again.. really really nit

Reply:
> We are trying to check if gotErr and wantErr are either both Nil or not nil.

Yes, basically we are trying to verify if we got an error when we expect one, and we dont get an error when we don't expect one. This is a fairly common practice in Go code wherein `gotErr` means `did I get an error?` and `wantErr` means `do i expect to see an error?` here.  It is also common practice for `gotErr` and `wantErr` to also contain the actual error seen by the test and the actual error expected by the test. These are used interchangeably.

I think the more you see this pattern, the more you will feel comfortable with it. 

If I have to rename it as `gotNilErr`, then it actually should be `gotNilErr := (err == nil)`, and that is not what we are doing here. So, I will leave this as is.